src/python/pants/backend/go/goals/check.py

```diff
@@ -24,7 +24,7 @@ class GoCheckFieldSet(FieldSet):
 
 class GoCheckRequest(CheckRequest):
     field_set_type = GoCheckFieldSet
-    name = "go compile"
+    name = "go-compile"
```

Comment:
(Non-blocking) So this is the odd duckling. If I look at the code, it seems like the `name` classvar is only used by the `debug_hint` (which could easily be changed, but idk what it's for) and then re-used for DRYness in each rule.

So these could be nixed in every class, the rules could uses the subsystem `options_scope` directly, and then this one instance would just hardcode in the rule.

Although none of this codifies the convention you're going for, which could be ideal but idk what that'd look like :smile: